metricbeat/module/couchdb/server/data.go

```diff
@@ -0,0 +1,131 @@
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package server
+
+import (
+	"encoding/json"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// Server type defines all fields of the Server Metricset
+type Server struct {
+	Httpd               Httpd               `json:"httpd"`
+	HttpdRequestMethods HttpdRequestMethods `json:"httpd_request_methods"`
+	HttpdStatusCodes    HttpdStatusCodes    `json:"httpd_status_codes"`
+	Couchdb             Couchdb             `json:"couchdb"`
+}
+
+// Httpd type defines httpd fields of the Server Metricset
+type Httpd struct {
+	ViewReads                General `json:"view_reads"`
+	BulkRequests             General `json:"bulk_requests"`
+	ClientsRequestingChanges General `json:"clients_requesting_changes"`
+	TemporaryViewReads       General `json:"temporary_view_reads"`
+	Requests                 General `json:"requests"`
+}
+
+// HttpdRequestMethods type defines httpd requests methods fields of the Server Metricset
+type HttpdRequestMethods struct {
+	Copy   General `json:"COPY"`
+	Head   General `json:"HEAD"`
+	Post   General `json:"POST"`
+	Delete General `json:"DELETE"`
+	Get    General `json:"GET"`
+	Put    General `json:"PUT"`
+}
+
+// HttpdStatusCodes type defines httpd status codes fields of the Server Metricset
+type HttpdStatusCodes struct {
+	Num200 General `json:"200"`
+	Num201 General `json:"201"`
+	Num202 General `json:"202"`
+	Num301 General `json:"301"`
+	Num304 General `json:"304"`
+	Num400 General `json:"400"`
+	Num401 General `json:"401"`
+	Num403 General `json:"403"`
+	Num404 General `json:"404"`
+	Num405 General `json:"405"`
+	Num409 General `json:"409"`
+	Num412 General `json:"412"`
+	Num500 General `json:"500"`
+}
+
+// Couchdb type defines couchdb fields of the Server Metricset
+type Couchdb struct {
+	OpenOsFiles     General `json:"open_os_files"`
+	OpenDatabases   General `json:"open_databases"`
+	AuthCacheHits   General `json:"auth_cache_hits"`
+	RequestTime     General `json:"request_time"`
+	DatabaseReads   General `json:"database_reads"`
+	DatabaseWrites  General `json:"database_writes"`
+	AuthCacheMisses General `json:"auth_cache_misses"`
+}
+
+// General type defines common fields of the Server Metricset
+type General struct {
+	Current float64 `json:"current"`
+}
+
+func eventMapping(content []byte) common.MapStr {
+	var data Server
+	json.Unmarshal(content, &data)
```

Comment:
Check for errors here.

Reply:
Thanks for your review. I believe I've addressed all the feedback now.